services/venice-controller/src/main/java/com/linkedin/venice/pushmonitor/PushMonitorUtils.java

```diff
@@ -82,12 +85,33 @@ public static ExecutionStatusWithDetails getDaVinciPushStatusAndDetails(
         incompletePartition.add(partitionId);
       }
     }
+    boolean noDaVinciStatusReported = totalReplicaCount == 0;
+
+    // Report error if too many davinci instances are not alive for over 5 mins
+    if (totalReplicaCount > 6 && liveReplicaCount < 0.5 * totalReplicaCount) {
+      Long lastUpdateTime = storeVersionToDVCDeadInstanceTimeMap.get(topicName);
+      if (lastUpdateTime != null) {
+        if (lastUpdateTime + DAVINCI_DEAD_INSTANCE_REPORT_ERROR_WAIT < System.currentTimeMillis()) {
```

Comment:
Could we have a unit test about this? For now it does not seem to have any test for new logic..